Downloading the binaries through a proxy

ulrichard opened the pull request:
Downloading the binaries through a proxy if either HTTPS_PROXY or HTTP_PROXY environment variable is set.

closes #48

ulrichard commented on Cargo.toml:
```diff
@@ -18,7 +18,7 @@ home = "0.5.3"  # use same ver in build-dep
 
 [build-dependencies]
-ureq = "1.0"  # allows to keep MSRV 1.41.1
+ureq = "2.1"
```
I hope it's ok to use the same version as electrsd

RCasatta replied:
sorry for not giving an update on this but I prefer to put this in standby at the moment

RCasatta commented:
After #102 I would be inclined to accept this if you want to fix conflicts

ulrichard commented:
ok, rebased. 
In the meantime I changed the strategy with our CI a bit. Now https://github.com/RCasatta/electrsd/pull/51 would be more important to me, but it would still be good to have the proxy stuff included.

ulrichard commented:
I don't think the CI build error in 1.41.1 has anything to do with my changes.

RCasatta commented on build.rs:
```diff
@@ -114,10 +114,19 @@ mod download {
             println!("url:{}", url);
             let mut downloaded_bytes = Vec::new();
-            let resp = ureq::get(&url).call().unwrap();
-            assert_eq!(resp.status(), 200, "url {} didn't return 200", url);
```
I don't remember exactly why this assert is here but maybe it helps with reporting? Can we restore it?

RCasatta reviewed (commented):
> I don't think the CI build error in 1.41.1 has anything to do with my changes.

It doesn't `which` dep broke msrv with their new `4.4.0` version... Don't know what is best to do if pinning the dep or including a Cargo.lock, @kixunil what do you think?

> In the meantime I changed the strategy with our CI a bit. Now https://github.com/RCasatta/electrsd/pull/51 would be more important to me,

I'll do my best to tackle that one too

Kixunil commented on .github/workflows/test.yml:
```diff
@@ -83,6 +83,10 @@ jobs:
+      - uses: actions-rs/cargo@v1
+        with:
+          command: update 
+          args: -p which --precise 4.3.0
```
Yeah, this is one of two good ways I know of to resolve the MSRV problem.
